lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -257,11 +274,26 @@ class MockTimers {
           );
         },
         setInterval: () => {
-          this.#realSetInterval = globalThis.setInterval;
-          this.#realClearInterval = globalThis.clearInterval;
-          this.#realTimersSetInterval = nodeTimers.setInterval;
-          this.#realTimersClearInterval = nodeTimers.clearInterval;
-          this.#realPromisifiedSetInterval = nodeTimersPromises.setInterval;
+          this.#realSetInterval = ObjectGetOwnPropertyDescriptor(
+            globalThis,
+            'setInterval',
+          );
+          this.#realClearInterval = ObjectGetOwnPropertyDescriptor(
+            globalThis,
+            'clearInterval',
+          );
+          this.#realTimersSetInterval = ObjectGetOwnPropertyDescriptor(
+            nodeTimers,
+            'setInterval',
+          );
+          this.#realTimersClearInterval = ObjectGetOwnPropertyDescriptor(
+            nodeTimers,
+            'clearInterval',
+          );
+          this.#realPromisifiedSetInterval = ObjectGetOwnPropertyDescriptor(
+            nodeTimersPromises,
+            'setInterval',
+          );
```

Comment:
since this block is copy-pasted multiple times, should it be its own private method?

Reply:
I thought so, but it'd be a function with two parameters that only call the property Descriptor. The other way would be making a tuple and iterating over it to call the getOwnProperty but I wonder if it'd only add unnecessarily complexity. WDYT?

Reply:
i was assuming you'd just move these 6 assignments into a no-argument private method, and call it in the three places that currently do this.

Reply:
got it! done!